storage: TestLargeUnsplittableRangeReplicate failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/bc3342ea2810c88f88b243491f41af581a54ba10

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1463109&tab=buildLog

```
W190831 06:51:28.972185 1177197 ts/db.go:197  [n4,ts-poll] node unavailable; try another peer
W190831 06:51:28.980044 1179783 storage/store.go:1546  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E190831 06:51:28.971791 3303758 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190831 06:51:29.001490 3303747 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190831 06:51:29.010305 3303723 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W190831 06:51:29.010787 1175201 storage/store.go:1546  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190831 06:51:29.037500 1174061 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
W190831 06:51:29.037863 1174061 ts/db.go:197  [n2,ts-poll] node unavailable; try another peer
E190831 06:51:29.050369 3303880 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W190831 06:51:29.065535 1175231 ts/db.go:194  [n3,ts-poll] error writing time series data: node unavailable; try another peer
W190831 06:51:29.073518 1175231 ts/db.go:197  [n3,ts-poll] node unavailable; try another peer
W190831 06:51:29.092705 1177206 storage/store.go:1546  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190831 06:51:29.093858 1179783 storage/store.go:1546  [n5,s5,r1/5:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E190831 06:51:29.108454 3303905 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W190831 06:51:29.116251 1175201 storage/store.go:1546  [n3,s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
E190831 06:51:29.128220 3303915 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190831 06:51:29.159212 3303921 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190831 06:51:29.161985 3303932 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190831 06:51:29.171323 3303951 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190831 06:51:29.177160 3303934 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190831 06:51:29.231130 3304007 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190831 06:51:29.232952 3304008 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestLargeUnsplittableRangeReplicate (730.44s)
    require.go:794: 
        	Error Trace:	replicate_queue_test.go:418
        	Error:      	Received unexpected error:
        	            	pq: unable to select removal target from [(n4,s4):?]; current replicas [(n1,s1):1 (n2,s2):2 (n4,s4):3 (n3,s3):4]: could not select an appropriate replica to be removed
        	Test:       	TestLargeUnsplittableRangeReplicate

```

SHA: https://github.com/cockroachdb/cockroach/commits/e186c663735a9210a92a81f36f30aaabadc2754c

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1474873&tab=buildLog

```
E190908 06:24:16.362593 3694804 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190908 06:24:16.368906 3694737 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190908 06:24:16.370222 3693985 util/stop/stopper.go:542  quiescing; tasks left:
2      node.Node: batch
1      ts.poller: poll
I190908 06:24:16.416107 3693985 util/stop/stopper.go:542  quiescing; tasks left:
1      ts.poller: poll
I190908 06:24:16.431389 3693981 util/stop/stopper.go:542  quiescing; tasks left:
1      ts.poller: poll
1      node.Node: batch
E190908 06:24:16.436960 3694818 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W190908 06:24:16.447342 1311582 storage/store.go:1546  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190908 06:24:16.448785 3693981 util/stop/stopper.go:542  quiescing; tasks left:
1      ts.poller: poll
I190908 06:24:16.481358 1318977 rpc/nodedialer/nodedialer.go:144  [n4] unable to connect to n5: failed to connect to n5 at 127.0.0.1:42571: context canceled
E190908 06:24:16.500100 3694954 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W190908 06:24:16.532011 1321715 ts/db.go:194  [n5,ts-poll] error writing time series data: node unavailable; try another peer
W190908 06:24:16.532455 1321715 ts/db.go:197  [n5,ts-poll] node unavailable; try another peer
W190908 06:24:16.543857 1311621 ts/db.go:194  [n1,ts-poll] error writing time series data: node unavailable; try another peer
W190908 06:24:16.544260 1311621 ts/db.go:197  [n1,ts-poll] node unavailable; try another peer
E190908 06:24:16.554964 3694998 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190908 06:24:16.582112 3694993 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190908 06:24:16.585454 3694949 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190908 06:24:16.590970 3695017 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190908 06:24:16.593294 3695001 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190908 06:24:16.594788 3694994 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W190908 06:24:16.665074 1319573 server/node.go:788  [n4,summaries] error recording status summaries: node unavailable; try another peer
E190908 06:24:16.688029 3695052 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
--- FAIL: TestLargeUnsplittableRangeReplicate (835.43s)
    require.go:794: 
        	Error Trace:	replicate_queue_test.go:418
        	Error:      	Received unexpected error:
        	            	pq: unable to select removal target from [(n5,s5):?]; current replicas [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n3,s3):5]: could not select an appropriate replica to be removed
        	Test:       	TestLargeUnsplittableRangeReplicate

```

SHA: https://github.com/cockroachdb/cockroach/commits/1b75c93fd2dc6c9575454e273040b0757f64a8c0

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1477368&tab=buildLog

```
W190910 06:30:13.558541 1244207 storage/store.go:1546  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190910 06:30:13.584896 1248157 storage/store.go:1546  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190910 06:30:13.602757 1248064 ts/db.go:194  [n2,ts-poll] error writing time series data: node unavailable; try another peer
W190910 06:30:13.603131 1248064 ts/db.go:197  [n2,ts-poll] node unavailable; try another peer
E190910 06:30:13.609056 4716835 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190910 06:30:13.609548 1244205 rpc/nodedialer/nodedialer.go:144  [n1] unable to connect to n5: failed to connect to n5 at 127.0.0.1:44077: context canceled
E190910 06:30:13.611324 4716836 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190910 06:30:13.611654 1244205 rpc/nodedialer/nodedialer.go:144  [n1] unable to connect to n2: failed to connect to n2 at 127.0.0.1:44139: context canceled
W190910 06:30:13.615852 1244207 storage/store.go:1546  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190910 06:30:13.634157 1254032 ts/db.go:194  [n5,ts-poll] error writing time series data: node unavailable; try another peer
W190910 06:30:13.634517 1254032 ts/db.go:197  [n5,ts-poll] node unavailable; try another peer
W190910 06:30:13.644271 1244258 ts/db.go:194  [n1,ts-poll] error writing time series data: node unavailable; try another peer
W190910 06:30:13.644609 1244258 ts/db.go:197  [n1,ts-poll] node unavailable; try another peer
E190910 06:30:13.647751 4716997 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
W190910 06:30:13.648625 1244134 gossip/gossip.go:1517  [n1] no incoming or outgoing connections
W190910 06:30:13.652868 1249638 ts/db.go:194  [n3,ts-poll] error writing time series data: node unavailable; try another peer
W190910 06:30:13.653167 1249638 ts/db.go:197  [n3,ts-poll] node unavailable; try another peer
E190910 06:30:13.653309 4717061 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190910 06:30:13.666445 1251319 rpc/nodedialer/nodedialer.go:144  [n4] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45035: context canceled
E190910 06:30:13.667845 4716976 vendor/google.golang.org/grpc/pickfirst.go:61  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190910 06:30:13.668133 1251319 rpc/nodedialer/nodedialer.go:144  [n4] unable to connect to n2: failed to connect to n2 at 127.0.0.1:44139: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (1100.68s)
    replicate_queue_test.go:451: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3,5}
        goroutine 1243900 [running]:
        runtime/debug.Stack(0xc000446140, 0x5fbe880, 0xc002049450)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x60ccea0, 0xc011b9d800, 0xc000446140)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:37 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc011b9d800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:451 +0xb5e
        testing.tRunner(0xc011b9d800, 0x522a768)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

--- a/pkg/storage/replicate_queue_test.go
+++ b/pkg/storage/replicate_queue_test.go
@@ -29,6 +29,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage/storagepb"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/gogo/protobuf/proto"
@@ -372,8 +373,8 @@ func toggleSplitQueues(tc *testcluster.TestCluster, active bool) {
 func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	if testing.Short() {
-		t.Skip("short flag - #38565")
+	if testing.Short() || testutils.NightlyStress() || util.RaceEnabled {
+		t.Skip("https://github.com/cockroachdb/cockroach/issues/38565")
 	}
 	ctx := context.Background()
 

